src/Ledger/Foreign/HSLedger/Certs.agda

```diff
@@ -0,0 +1,94 @@
+module Ledger.Foreign.HSLedger.Certs where
+
+open import Ledger.Foreign.HSLedger.BaseTypes
+open import Ledger.Foreign.HSLedger.Enact
+open import Ledger.Foreign.HSLedger.Gov
+
+import Ledger.Foreign.LedgerTypes as F
+import Foreign.Haskell.Pair as F
+
+open import Ledger.Certs.Properties HSGovStructure
+
+instance
+  _ = Convertible-Refl {String}
+
+  Convertible-PoolParams : Convertible PoolParams F.PoolParams
+  Convertible-PoolParams = λ where
+    .to → to ∘ PoolParams.rewardAddr
+    .from x → record { rewardAddr = from x }
+
+  Convertible-DCert : Convertible DCert F.TxCert
+  Convertible-DCert = autoConvertible
+
+  Convertible-TxBody : Convertible TxBody F.TxBody
+  Convertible-TxBody = λ where
+    .to txb → let open TxBody txb in record
+      { txins  = to txins
+      ; refInputs  = to refInputs
+      ; txouts = to txouts
+      ; txfee  = txfee
+      ; txvldt = to txvldt
+      ; txsize = txsize
+      ; txid   = to txid
+      ; collateral = to collateral
+      ; reqSigHash = to reqSigHash
+      ; scriptIntHash = nothing
+      ; txcerts = to txcerts
+      }
+    .from txb → let open F.TxBody txb in record
+      { txins      = from txins
+      ; refInputs  = from refInputs
+      ; txouts     = from txouts
+      ; txcerts    = from txcerts
+      ; mint       = ε -- tokenAlgebra only contains ada atm, so mint is surely empty
+      ; txfee      = txfee
+      ; txvldt     = from txvldt
+      ; txwdrls    = ∅
+      ; txup       = nothing
+      ; txADhash   = nothing
+      ; netwrk     = nothing
+      ; txsize     = txsize
+      ; txid       = from txid
+      ; txvote     = []
+      ; txprop     = []
+      ; txdonation = ε
+      ; collateral    = from collateral
+      ; reqSigHash    = from reqSigHash
+      ; scriptIntHash = nothing
+      }
+
+  Convertible-Tx : Convertible Tx F.Tx
```

Comment:
I'm a bit surprised that `Tx` and `TxBody` are in this file. Does it have to be here, or can it be moved to something more natural?

Reply:
Turns out it doesn't really need to be here. I've moved it to `HSLedger.BaseTypes`